module/command_tree_generator.py

```diff
@@ -0,0 +1,92 @@
+import json
+import inspect
+
+import common.frames
+
+from module.command_node import NodeType, Node
+from module.frame_functions import get_frames_with_description
+
+
+def add_command(parent, name, parameters, description) -> None:
+    """
+    Add one command with all the required information
+    """
+    target_node = Node(
+        name,
+        NodeType.COMMAND,
+        parameter_list=parameters,
+        command_info=description
+    )
+    target_node.set_parent(parent)
+    parent[target_node.name] = target_node
+
+def add_frame_commands(root_node, mod=common.frames) -> None:
+    """
+    Adds all commands from the mod file
+    """
+    #check if robot already exists, otherwise create it
+    if not 'ROBOT' in root_node:
+        root_node["ROBOT"] = Node("ROBOT", NodeType.CATEGORY, root_node)
+    current_node = root_node["ROBOT"]
+
+    commands = get_frames_with_description(mod)
+
+    #add all commands to root node
+    for command in commands:
+        name = command[0][5:]
```

Comment:
From the code alone I can't see why you are indexing like this. A comment would clarify that